From: Zebediah Figura <zfigura@codeweavers.com>
Subject: [PATCH 2/2] server: Remove a redundant NULL check for async->fd.
Message-Id: <20210821192951.3744676-2-zfigura@codeweavers.com>
Date: Sat, 21 Aug 2021 14:29:51 -0500
In-Reply-To: <20210821192951.3744676-1-zfigura@codeweavers.com>
References: <20210821192951.3744676-1-zfigura@codeweavers.com>

If the async has not been terminated yet, async->fd cannot be NULL.

Signed-off-by: Zebediah Figura <zfigura@codeweavers.com>
---
 server/async.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/async.c b/server/async.c
index dbd4859edc4..442d98d8c7a 100644
--- a/server/async.c
+++ b/server/async.c
@@ -448,7 +448,7 @@ restart:
     LIST_FOR_EACH_ENTRY( async, &process->asyncs, struct async, process_entry )
     {
         if (async->status != STATUS_PENDING) continue;
-        if ((!obj || (async->fd && get_fd_user( async->fd ) == obj)) &&
+        if ((!obj || (get_fd_user( async->fd ) == obj)) &&
             (!thread || async->thread == thread) &&
             (!iosb || async->data.iosb == iosb))
         {

-- 
2.30.2

